Improve documentation about environment variables being used

pravarag opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind documentation
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
As part of this PR, we are adding documentation for environment variables that are used to configure kube-scheduler-simulator
#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #106 

#### Special notes for your reviewer:


/label tide/merge-method-squash

pravarag commented:
> Thanks for opening the PR. Could you move this doc to README so that users don't miss it?

Sure, do you want me to put the entire contents in README itself or provide a link to this new document in the README file?

sanposhiho commented:
> provide a link to this new document in the README file?

👍 nice idea. just adding a link to README seems to be fine.

sanposhiho commented:
@pravarag 
Could you add the description about `KUBE_SCHEDULER_CONFIG_PATH` on `docs/env-variables.md` as well?
https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/131/files#r857065848

We already have the doc about this here:
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/README.md?plain=1#L32
But, I think it's better to move this to `docs/env-variables.md`.

sanposhiho commented:
And could you open this PR?

sanposhiho commented:
/retitle Improve documentation about environment variables being used

pravarag commented on .gitignore:
```diff
@@ -1 +1,7 @@
-bin/
\ No newline at end of file
+<<<<<<< HEAD
```
Apologies for the bad rebase, fixed with the latest push.

pravarag commented:
@sanposhiho I've updated the PR with some new env variables, kindly take a look and let me know for any changes/updates required.

sanposhiho commented:
@pravarag 
Please add link for docs/env-variables.md at somewhere on README.md.

pravarag commented:
> @pravarag Please add link for docs/env-variables.md at somewhere on README.md.

Done with the latest push.

196Ikuchil commented on docs/env-variables.md:
```diff
@@ -0,0 +1,20 @@
+# Different Environment Variables used
```
Maybe we should also describe about `KUBECONFIG`?
like this.
```
`KUBECONFIG`:The cluster from which you are importing is indicated by this environment variable.
  [Read more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
```

pravarag replied:
Done with the latest push. Thanks for mentioning that.

196Ikuchil commented on docs/env-variables.md:
```diff
@@ -0,0 +1,20 @@
+`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)
+
+`KUBE_SCHEDULER_CONFIG_PATH`: A KubeSchedulerConfiguration file can be
+passed via this environment varia
+ble and the simulator will start the
```
```suggestion
passed via this environment variable and the simulator will start the
```
On Github's markdown view, space was inserted into `variable`

pravarag replied:
Updated.

196Ikuchil commented on docs/env-variables.md:
```diff
@@ -0,0 +1,23 @@
+
+`FrontendURL`: This URL is on which kube-scheduler-simulator is reachable locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L9).
+
+`KUBECONFIG`: This variable is used to access the cluster which is
+imported locally. [Read
```
@pravarag thanks!
hmm, at least, in our simulator, this value is used to access the cluster when we'd like to import the cluster's resources. So, I think that it is not a `locally imported cluster` but a `cluster which will be imported locally`.
(But my English is not good, sorry...)
WDOT?

pravarag replied:
How about, 

```
This variable is used by kubectl tool to find the information it needs to access the cluster which will be imported locally and communicate with the API server of the cluster. 
```
I took some reference from the official Kubernetes docs.

pravarag replied:
Let me know your thoughts on the above

196Ikuchil replied:
`KUBECONFIG`
[In general simulator]: Currently not used.
[In Beta]: Used for settings indicating clusters to be imported. To be precise, the KubeConfig file of the cluster to be imported is specified so that simulator can read the resources in the specified cluster.

So, How is this?
```
`KUBECONFIG`: Currently not used. [Beta: Existing Cluster Importing] This variable is used to find the information it needs to access the cluster which will be imported to our simulator and communicate with the API server of the cluster. If you use the Beta feature, you need to set the `Kubeconfig` file path of the cluster which will be imported. [Read more]
```

pravarag replied:
Updated.

196Ikuchil commented on docs/env-variables.md:
```diff
@@ -0,0 +1,23 @@
+
+`KUBE_SCHEDULER_CONFIG_PATH`: A KubeSchedulerConfiguration file can be
+passed via this environment variable and the simulator will start the
+scheduler with that configuration.
```
Also, could you add this description?
```suggestion
scheduler with that configuration. [Beta: Existing Cluster Importing] The same is true for scheduler settings, even for beta features. This is because scheduler settings cannot be imported directly from an existing cluster.
```

pravarag replied:
Done. updated as per your suggestions with the latest push.

196Ikuchil reviewed (commented):
@pravarag 
Thanks for your reply!
I left some comments. Please check it~

196Ikuchil commented:
Thank you!
LGTM about the perspective of the #131 .

ping @sanposhiho 

/lgtm

sanposhiho commented on .gitignore:
```diff
@@ -1 +1,3 @@
-bin/
\ No newline at end of file
+bin/
+.vscode/
+*~
```
Please revert non-related change

pravarag replied:
Updated.

sanposhiho commented on README.md:
```diff
@@ -81,6 +81,8 @@ It starts etcd and simulator-server locally.
 
 To run the frontend, please see [README.md](./web/README.md) on ./web dir.
 
+Read more about differnet environment variables being used in simulator server [here.](./docs/env-variables.md)
```
```suggestion
Read more about environment variables being used in simulator server [here.](./docs/env-variables.md)
```

sanposhiho replied:
And please move it to L48 since these configurations is not only related to simulator frontend.

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+
+`FrontendURL`: This URL is on which kube-scheduler-simulator is reachable locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L9).
+
+`KUBECONFIG`: Currently not used. [Beta: Existing Cluster Importing]
+This variable is used to find the information it needs to access the
+cluster which will be imported to our simulator and communicate with
+the API server of the cluster. If you use the Beta feature, you need
+to set the `Kubeconfig` file path of the cluster which will be
+imported. [Read more]. [Read
+more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
```
```suggestion
`KUBECONFIG`: This is for the beta feature "Existing Cluster Importing" 
and the feature is enabled when you set the value.
Currently, the simulator tries to import ALL resources related to scheduling from your cluster,
so, please be careful (or avoid) using it in clusters with large amounts of resources. 
This variable is used to find KubeConfig it needs to access your
cluster to import resources to our simulator. 
```

pravarag replied:
Made a small change in the last two lines: `Kubeconfig required to access your cluster for importing resources to scheduler simulator`. Let me know if this looks good.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+imported. [Read more]. [Read
+more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
+
+`KUBE_API_HOST`: This is the host on which the kube-apiserver serves
+for the simulator server.
```
```suggestion
`KUBE_API_HOST`: This is the host of kube-apiserver which the simulator starts internally.
```

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+`KUBE_API_HOST`: This is the host on which the kube-apiserver serves
+for the simulator server.
+
+`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)
```
This `KUBE_API_SERVER_URL` is env for frontend. And I think we should write the doc about env for frontend in another section to avoid confusion.

Please create another section (like `## For web UI`) and move `KUBE_API_SERVER_URL` in the section

pravarag replied:
Done, thanks for mentioning that.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+
+`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)
+
+`KUBE_SCHEDULER_CONFIG_PATH`: A KubeSchedulerConfiguration file can be
+passed via this environment variable and the simulator will start the
+scheduler with that configuration. [Beta: Existing Cluster Importing] The same is true for scheduler settings, even for beta features. This is because scheduler settings cannot be imported directly from an existing cluster.
```
```suggestion
`KUBE_SCHEDULER_CONFIG_PATH`: The path to a KubeSchedulerConfiguration file. 
If passed, the simulator will start the scheduler with that configuration.
Or, if you use web UI, you can change the configuration from the web UI as well..
```

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+# Different Environment Variables used
+
+This page describes about the different environment variables that are
+used to configure kube-scheduler-simulator.
```
```suggestion
This page describes about the environment variables that are
used to configure kube-scheduler-simulator.

You can refer [docker-compose.yml](./docker-compose.yml) as an example use.
```

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+
+`Port`: This is the port number on which kube-scheduler-simulator is started and it's value is 1212 which is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L7)
+
+`EtcdURL`: This is the URL for kube-scheduler-simulator etcd which runs locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L8)
```
```suggestion
`EtcdURL`: This is the URL for etcd. 
The simulator run kube-apiserver internally, and the kube-apiserver uses this etcd.
```

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+This page describes about the different environment variables that are
+used to configure kube-scheduler-simulator.
+
+`Port`: This is the port number on which kube-scheduler-simulator is started and it's value is 1212 which is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L7)
```
```suggestion
`Port`: This is the port number on which kube-scheduler-simulator server is started.
```

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,27 @@
+
+`EtcdURL`: This is the URL for kube-scheduler-simulator etcd which runs locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L8)
+
+`FrontendURL`: This URL is on which kube-scheduler-simulator is reachable locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L9).
```
```suggestion
`FrontendURL`: This URL represents the URL web UI started.
The simulator and internal kube-apiserver set this origin as the allowed origin for CROS.
```

sanposhiho replied:
Sorry, it's already renamed to CORS_ALLOWED_ORIGIN_LIST. Please rename
https://github.com/kubernetes-sigs/kube-scheduler-simulator/commit/c922c36d0f5678c9f5a9c9573aa333bbed91b43e

pravarag replied:
Updated.

sanposhiho commented:
And please add the description about the default value for each variable. And if the environment variable is required one, you may want to add the note about it.

sanposhiho reviewed (commented):
the description for EXTERNAL_IMPORT_ENABLED is missing

pravarag commented:
> the description for EXTERNAL_IMPORT_ENABLED is missing

Thanks for a detailed review above. I've added the mentioned variable and also added `(required)` in front of mandatory variables. Let me know if this looks good or any other changes are required.
@sanposhiho @196Ikuchil

pravarag commented:
@sanposhiho @196Ikuchil let me know if any further changes are required for this PR :)

sanposhiho commented:
Ahhh, sorry, I missed the notification.
I will check it this weekend.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,35 @@
+simulator and internal kube-apiserver set the origin as the allowed
+origin for `CORS_ALLOWED_ORIGIN_LIST`.
+
+`KUBECONFIG`: (required) This is for the beta feature "Existing cluster
```
It's not required env.

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,35 @@
+`KUBE_API_HOST`: (required) This is the host of kube-apiserver which the
+simulator starts internally.
+
+`KUBE_SCHEDULER_CONFIG_PATH`: (required) The path to a KubeSchedulerConfiguration
```
It's also not required.

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,35 @@
+Importing". This variable is used to find Kubeconfig required to
+access your cluster for importing resources to scheduler simulator.
+
+`KUBE_API_HOST`: (required) This is the host of kube-apiserver which the
```
same. not required one.

pravarag replied:
Updated

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,35 @@
+Please refer [docker-compose.yml](./docker-compose.yml) as an example
+use.
+
+`Port`: (required) This is the port number on which kube-scheduler-simulator
+server is started. It's default value is `1212`.
```
No default value on it.
```suggestion
`Port`: (required) This is the port number on which kube-scheduler-simulator
server is started.
```

pravarag replied:
Updated.

sanposhiho commented on docs/env-variables.md:
```diff
@@ -0,0 +1,35 @@
+`Port`: (required) This is the port number on which kube-scheduler-simulator
+server is started. It's default value is `1212`.
+
+`EtcdURL`: (required) This is the URL for etcd. The simulator runs kube-apiserver
+internally, and the kube-apiserver uses this etcd. It's default value is `http://simulator-etcd:2379`.
```
No default value on it. And I guess you mean `KUBE_SCHEDULER_SIMULATOR_ETCD_URL`.

```suggestion
`KUBE_SCHEDULER_SIMULATOR_ETCD_URL`: (required) This is the URL for etcd. The simulator runs kube-apiserver
internally, and the kube-apiserver uses this etcd.
```

pravarag replied:
Updated.

sanposhiho reviewed (commented):
Sorry for too late reviews 🙇 
Btw, please feel free to ping me on Kubernetes slack. (I often miss the notification from GitHub because I receive them a lot every day. 😓 )

pravarag commented:
> Sorry for too late reviews bow Btw, please feel free to ping me on Kubernetes slack. (I often miss the notification from GitHub because I receive them a lot every day. sweat )

Thanks for the review @sanposhiho , I've updated the code as per the suggested changes :)

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,36 @@
+Please refer [docker-compose.yml](./docker-compose.yml) as an example
+use.
+
+`Port`: (required) This is the port number on which kube-scheduler-simulator
```
```suggestion
`PORT`: (required) This is the port number on which kube-scheduler-simulator
```

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,36 @@
+etcd. The simulator runs kube-apiserver internally, and the
+kube-apiserver uses this etcd.
+
+`FrontendURL`: This URL represents the URL web UI started. The
```
It's already renamed to CORS_ALLOWED_ORIGIN_LIST.

pravarag replied:
Updated.

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,36 @@
+Importing". This variable is used to find Kubeconfig required to
+access your cluster for importing resources to scheduler simulator.
+
+`KUBE_API_HOST`: This is the host of kube-apiserver which the
+simulator starts internally.
```
```suggestion
`KUBE_API_HOST`: This is the host of kube-apiserver which the
simulator starts internally. Its default value is `3131`.
```

sanposhiho replied:
And `KUBE_API_PORT` env should be mentioned as well. `KUBE_API_PORT` is the port of kube-apiserver.

pravarag replied:
Updated.

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,36 @@
+configuration.  Or, if you use web UI, you can change the
+configuration from the web UI as well.
+
+`ExternalImportEnabled`: This variable indicates whether the simulator will import resources from an existing cluster or not.
```
```suggestion
`ExternalImportEnabled`: This variable indicates whether the simulator will import resources from an existing cluster or not. Note that this is still a beta feature.
```

pravarag replied:
Updated.

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,36 @@
+`ExternalImportEnabled`: This variable indicates whether the simulator will import resources from an existing cluster or not.
+
+## For Web UI
+
```
Also, for web ui, ALPHA_TABLE_VIEWS and BASE_URL should be mentioned.
They do the followings. I am not a good English speaker, feel free to change the wording as you like.

BASE_URL → the URL for the simulated server.
ALPHA_TABLE_VIEWS → The value is 0 or 1. (0 means disabled, 1 means enabled.) Enable the alpha feature "table view". You can see the resource status in the table.

pravarag replied:
Updated as suggested above.

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,46 @@
+Importing". This variable is used to find Kubeconfig required to
+access your cluster for importing resources to scheduler simulator.
+
+`KUBE_API_HOST`: This is the host of kube-apiserver which the
+simulator starts internally. It's default value is `3131`.
+
+`KUBE_API_PORT`: This is the port of kube-apiserver.
```
Sorry, what I suggested before was wrong 😓 

```suggestion
`KUBE_API_HOST`: This is the host of kube-apiserver which the
simulator starts internally. It's default value is `127.0.0.1`.

`KUBE_API_PORT`: This is the port of kube-apiserver.  It's default value is `3131`.
```

pravarag replied:
Done. Updated with latest push.

196Ikuchil commented on README.md:
```diff
@@ -73,6 +73,8 @@ If you want to use your custom plugins as out-of-tree plugins in the simulator,
 ## Getting started
 
+Read more about environment variables being used in simulator server
+[here.](./docs/env-variables.md)
```
```suggestion
[here.](./simulator/docs/env-variables.md)
```

pravarag replied:
Updated.

196Ikuchil commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,47 @@
+This page describes about the different environment variables that are
+used to configure kube-scheduler-simulator. 
+
+Please refer [docker-compose.yml](./docker-compose.yml) as an example
```
```suggestion
Please refer [docker-compose.yml](./../../docker-compose.yml) as an example
```
This link maybe not working.

pravarag replied:
Updated.

196Ikuchil commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,47 @@
+
+## For Web UI
+
+`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)
+
+`BASE_URL`: This is the URL for the kube-scheduler-simulator server.
+
+`ALPHA_TABLE_VIEWS`: This variable enables the alpha feature `table
+view`. It's value is either 0 or 1 (0 meaning disalbed, 1 meaning
+enabled). We can see the resource status in the table.
```
```suggestion
`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26). Its default value is `http://localhost:3131`.

`BASE_URL`: This is the URL for the kube-scheduler-simulator server. Its default value is `http://localhost:1212`.

`ALPHA_TABLE_VIEWS`: This variable enables the alpha feature `table
view`. It's value is either 0 or 1 (0 meaning disabled, 1 meaning
enabled). We can see the resource status in the table. Its default value is `0`.
```

sanposhiho replied:
And please remove `it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26).`

pravarag replied:
Updated.

196Ikuchil reviewed (commented):
Hi, I'm sorry to keep repeating myself.
Please check these nits.

@sanposhiho I'd like to merge this PR as soon as possible. A new feature of the simulator is added, and we may be required to modify this PR every time?

sanposhiho commented on simulator/docs/env-variables.md:
```diff
@@ -0,0 +1,47 @@
+access your cluster for importing resources to scheduler simulator.
+
+`KUBE_API_HOST`: This is the host of kube-apiserver which the
+simulator starts internally. It's default value is `127.0.0.1`.
```
```suggestion
simulator starts internally. Its default value is `127.0.0.1`.
```

pravarag replied:
Updated.

pravarag commented:
Apologies for the delayed response, I've updated as per review comments @sanposhiho @196Ikuchil .

sanposhiho reviewed (approved):
/approve

Looks good to me. Thanks @pravarag !!

@196Ikuchil any thought? If no, please `/lgtm` 🙏

196Ikuchil commented:
Thank you for a very very long time!
/lgtm
